TribalWars/src/Infrastructure/Configuration/ArmyUnitRequiredMaterialsConfig.cs

```diff
@@ -0,0 +1,27 @@
+﻿using ApplicationCore.Entities;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Infrastructure.Configuration
+{
+    public class ArmyUnitRequiredMaterialsConfig : IEntityTypeConfiguration<ArmyUnitRequiredMaterial>
+    {
+        public void Configure(EntityTypeBuilder<ArmyUnitRequiredMaterial> builder)
+        {
+            builder.HasKey(i => i.Id);
+
+            builder.HasOne<ArmyUnitType>(a => a.ArmyUnitType)
+                .WithMany(x => x.ArmyUnitRequiredMaterials)
+                .HasForeignKey(x => x.ArmyUnitTypeId);
+
+            builder.HasOne<Material>(m => m.Material)
+                .WithMany(m => m.ArmyUnitRequiredMaterials)
+                .HasForeignKey(x => x.MaterialId);
```

Comment:
for information only:
for better understanding for the future, better way of naming variables in arrow function is to use meaningful var, which describe objects which we have inside
e.g. 
```
 builder.HasOne<ArmyUnitType>(requirements => requirements.ArmyUnitType)**
                .WithMany(unitType => unitType.ArmyUnitRequiredMaterials)
                .HasForeignKey(requirement => requirement.ArmyUnitTypeId)
```

Reply:
we use this approach to better understand what the action is about

Reply:
I leave you to consider whether you will change it or not